hw7/functional.py

```diff
@@ -0,0 +1,58 @@
+def sequential_map(*args):
+    container = list(args[-1])
+    functions = args[0:(len(args)-1)]
+    for function in functions:
+        container = list(map(function, container))
+    return container
+
+def consensus_filter(*args):
+    container = list(args[-1])
+    functions = args[0:(len(args)-1)]
+    for function in functions:
+        container = list(filter(function, container))
+    return container
+
+def conditional_reduce(fun1, fun2, container):
+    # Создадим функцию reduce вместо импорта из functools    
+    def reduce(function, iterable, initializer=None):
+        it = iter(iterable)
+        if initializer is None:
+            value = next(it)
+        else:
+            value = initializer
+        for element in it:
+            value = function(value, element)
+        return value
+    # Проведем операции с нашей функцией
+    container_fun1 = list(filter(fun1, container))
+    container_fun2 = reduce(fun2,container_fun1)
+    return container_fun2
+
+def func_chain(*args):
+    def chain(value):
+        for function in args:
+            value = function(value)
+        return value
+    return chain
+
+def sequential_map2(*args):
+    container = list(args[-1])
+    functions = func_chain(*args[0:(len(args)-1)])
+    return functions(container)
```

Comment:
Логика `sequential_map` не должна была изменяться. Этот набор функций мы всё также должны применять к каждому значению из контейнера, а не ко всему сразу
```suggestion
def sequential_map2(*args):
    container = list(args[-1])
    functions = func_chain(*args[0:(len(args)-1)])
    return list(map(functions, container))
```

Reply:
![image](https://user-images.githubusercontent.com/48160953/207434479-c93069ad-0642-4cb2-b89d-1f02fbf6c812.png)

Reply:
![image](https://user-images.githubusercontent.com/48160953/207434555-1723642f-e9a0-4c2e-997b-2abe5e92efa0.png)